File: 1003-check-if-word-is-valid-after-substitutions/1003-check-if-word-is-valid-after-substitutions.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let chars = [];
    for(let c of s) {
        chars.push(c);
        const n = chars.length;
        if(n >= 3 && chars[n-3] == 'a' && chars[n-2] == 'b' && chars[n-1] == 'c') {
            chars.pop(); chars.pop(); chars.pop();
        }
    }
    return chars.length == 0;
};

File: 1003-check-if-word-is-valid-after-substitutions/1003-check-if-word-is-valid-after-substitutions.py
class Solution(object):
    def isValid(self, s):
        chars = []
        for c in s:
            chars.append(c)
            n = len(chars)
            if n >= 3 and chars[n-3] == 'a' and chars[n-2] == 'b' and chars[n-1] == 'c':
                chars.pop()
                chars.pop()
                chars.pop()
        return len(chars) == 0